### PR TITLE
Reference guide document update

### PR DESCRIPTION
**Pull request by tatiana**
Originally created by @mikeshwe in the PR: https://github.com/astronomer/astro-sdk/pull/394
I created a new branch/PR so we can see the tests running in the CI smoothly, and so I could contribute, adding some minor fixes related to the 0.9 release.

### CODE REVIEW

File: REFERENCE.md [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+  Astro SDK Python :rocket:
+</h1>
+  <h3 align="center">
+  Your new Airflow DAG writing experience. Maintained with ❤️ by Astronomer.
+</h3>
+<br/>
+
+# :mega: Disclaimer :mega:
+```
+This project is in a previous release. Interfaces may change with feedback from early users.
```
Comment (dimberman) [inline REFERENCE.md:16]:
```suggestion
This project is in a preview release. Interfaces may change with feedback from early users.
```

File: REFERENCE.md [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+with dag:
+    my_table = my_first_sql_transformation(
+        input_table=Table(
+            name="foo", conn_id="postgres_conn", metadata=Metadata(schema="postgres")
+        )
+    )
+    my_second_sql_transformation(my_table)
+```
+
+### The TempTable Class
```
Comment (dimberman) [inline REFERENCE.md:117]:
```suggestion
### Temporary Tables
```

File: REFERENCE.md (hunk 1 above)
Comment (kaxil) [inline REFERENCE.md:16]:
```suggestion
This project is in a **preview** release. Interfaces may change with feedback from early users.
```

Something about "This project is in a **preview** release" still doesn't sound grammatically correct

**Review by dimberman — APPROVED** [review]

File: REFERENCE.md [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+
+    ```shell script
+    AIRFLOW__CORE__ENABLE_XCOM_PICKLING=True
+    ```
+
+# Usage
+
+## Setting Input and Output Tables
+
+Before we can complete any transformations, we need to define a way to get our tables in and out of Airflow. We can do this by defining either `Table` or `TempTable` objects in the `input_table` and `output_table` parameters of our table instantiations.
```
Comment (kaxil) [inline REFERENCE.md:72]: `TempTable` we don't have this object

**Review by kaxil — APPROVED** [review]

**Pull request merged**